### ISSUE TITLE
Procedure to make it work?

### ISSUE THREAD

**Issue by joel-porquet** [Hegemonia123/restoreToScreen#1]
Hi there, I really like the idea of this extension as I often have to disconnect/reconnect an external monitor to my laptop. Unfortunately, I just tried the extension and I wasn't able to make it work.

I enabled the extension, then I plugged in the external monitor and I moved some windows over to the monitor. I disconnected the monitor which brought all the windows back to the primary screen. I finally reconnected the external monitor, but no windows were automatically moved to the external monitor, they all stayed on the primary screen.

Is your extension meant to be used in another way? What's the typical scenario for which it works?

Thanks!

**Comment by Hegemonia123** [comment]
Hi!

There seems to be some problem with `EnabledByDefault` kwin script option. After installing new kwin script, I often have to disable the script and then re-enable it again to make it work.

However, I have noticed some issue in the script as well. It is likely caused by kwin itself moving windows from screen to another when external monitor is connected. I haven't came up with method to tell apart the screen change signals produced by kwin from the ones produced by user. Therefore, the scrip seems to work with some monitors and not with others. Even with the issue it is better with the script than without. At least in my case. 

I have an idea how I might be able to workaround the issue. I'll let you know if I succeed.
